operator/v1/types_console.go

```diff
@@ -143,8 +143,101 @@ type Capability struct {
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// Theme defines a theme type for the console UI.
+type Theme struct {
+	// type is a required field that specifies the type of the logo for the console UI.
+	// This field is an enum with valid values "Dark" and "Light".
+	// The type determines whether the logo is for the dark or light theme.
+	// When set to "Dark", the corresponding logo for the dark theme will be used in the console UI.
+	// When set to "Light", the corresponding logo for the light theme will be used in the console UI.
+	// When set to "All", the corresponding logo for both dark and light themes will be used in the console UI.
+	// +kubebuilder:validation:Enum="Dark";"Light";"All"
+	// +required
+	Type ThemeType `json:"type"`
+
+	// file is a required field that references a specific file within a ConfigMap in the openshift-config namespace.
+	// This field allows the console to locate and use the specified file containing a custom logo.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file
+	// The ConfigMap key should include the file extension for the dark or light theme so that the console serves the file with the correct MIME type.
+	// Recommended Masthead and About Modal logo specifications:
+	// Image size must be less than 1 MB due to constraints on the ConfigMap size
+	// Dimensions: Max height of 60px and max width of 200px
+	// SVG format preferred
+	// Recommended Favicon specifications:
+	// Dimensions: Max height of 16px and max width of 16px
+	// PNG format preferred
+	// +required
+	File configv1.ConfigMapFileReference `json:"file"`
+}
+
+// CustomLogo defines a configuration based on theme types for the console UI logo.
+type CustomLogo struct {
+	// type is a required field that specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+	// This field is an enum with valid values "Masthead" and "Favicon".
+	// Valid values are "Masthead" and "Favicon".
+	// When set to "Masthead", the logo will be used in the masthead or about modal of the console UI.
+	// When set to "Favicon", the logo will be used as the favicon of the console UI.
+	// +kubebuilder:validation:Enum="Masthead";"Favicon"
+	// +required
+	Type LogoType `json:"type"`
+
+	// themes is a required field that specifies the themes for the console UI logo.
+	// This field is a list of themes, each with a Type.
+	// Each theme determines whether the logo is for the dark or light theme of the console UI.
+	// The valid values for the Type field are "Dark" and "Light".
+	// Each item in the list must have a unique Type.
+	// When specified, there must be at least one entry and must not have greater than 2 entries.
```

Comment:
I would suggest that we also mention which logo will being rendered by default, if a logo for a available theme is missing, eg. what logo will get rendered for the Dark theme, if we only set custom logo for Light theme. I think in that case we should render the default OCP logo as @spadgett suggested.